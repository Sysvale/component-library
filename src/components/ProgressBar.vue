<template>
	<span
		id="progress-bar"
	>
		<div
			class="progress-bar-container"
		>
			<span
				v-if="showText"
				class="text-style"
			>
				{{ formatedPercentage }}
			</span>
			<div
				class="progress-bar-background"
				:style="backgroundColorStyle"
			> 
				<div
					class="progress-indicator"
					:style="progressIndicatorStyle"
				/>
			</div>
		</div>
	</span>
</template>

<script>
export default {
	props: {
		color: {
			type: String,
			default: '#2EB88D',
			description: 'Defines the color of the progress bar',
			required: false,
		},
		backgroundColor: {
			type: String,
			default: '#DDE2E7',
			description: 'Defines the background color of the progress bar',
			required: false,
		},
		percentage: {
			type: Number,
			default: 0,
			description: 'Defines the width of the progress Bar',
			required: true,
		},
		showText: {
			type: Boolean,
			default: false,
			description: 'Defines if the text will be displayed',
			required: false,
		}
	},

	computed: {
		backgroundColorStyle() {
			return {
				'--backgroundColor': this.backgroundColor,
			}
		},

		formatedPercentage() {
			return `${parseInt(this.percentage * 100)}%`;
		},

		progressIndicatorStyle() {
			return {
				'--width': this.formatedPercentage,
				'--color': this.color,
			}
		},
	},
};
</script>

<style>
#progress-bar .progress-indicator {
	border-radius: 80px;
	height: 7px;
	background-color: var(--color);
	width: var(--width);
}

#progress-bar .progress-bar-background {
	background-color: var(--backgroundColor);
	border-radius: 80px;
	width: 100%;
	height: 7px;
}

#progress-bar .progress-bar-container {
	display: flex;
	align-items: center;
	flex-direction: column;
}

#progress-bar .text-style {
	color: #4F6070;
	font-weight: 600;
}
</style>
