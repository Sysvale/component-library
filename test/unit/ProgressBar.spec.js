import { mount, createLocalVue } from '@vue/test-utils';
import BootstrapVue from 'bootstrap-vue';
import ProgressBar from '../../src/components/ProgressBar.vue';

const localVue = createLocalVue();
localVue.use(BootstrapVue);

test('Component is mounted properly', () => {
	const wrapper = mount(ProgressBar, {
		localVue,
		propsData: {
			percentage: 0.67,
		}
	});

	expect(wrapper).toMatchSnapshot();
});

describe("Colors are applied properly", () => {
	test("If the progress-bar color is the same that was passed via prop", () => {
		const color = '#23B1C7';
		
		const wrapper = mount(ProgressBar, {
			localVue,
			propsData: {
				color,
				percentage: 0.67,
			}
		});

		expect(wrapper.vm.progressIndicatorStyle['--color']).toBe(color);
	});

	test("If the background of the progress-bar is the same that was passed via prop", () => {
		const backgroundColor = '#000000';
		
		const wrapper = mount(ProgressBar, {
			localVue,
			propsData: {
				backgroundColor,
				percentage: 0.67,
			}
		});

		expect(wrapper.vm.backgroundColorStyle['--backgroundColor']).toBe(backgroundColor);
	});
});

test("If the size of the progress-bar changes properly", () => {
	const wrapper = mount(ProgressBar, {
		localVue,
		propsData: {
			percentage: 0.67,
		}
	});

	expect(wrapper.vm.progressIndicatorStyle['--width']).toBe("67%");
});

test("If the text is shown properly", () => {
	const wrapper = mount(ProgressBar, {
		localVue,
		propsData: {
			percentage: 0.53,
			showText: true,
		}
	});

	expect(wrapper.findAll('.text-style').length).toBe(1);
});
