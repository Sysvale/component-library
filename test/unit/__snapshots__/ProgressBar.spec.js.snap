// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component is mounted properly 1`] = `
<div class="progress-bar-container">
  <!---->
  <div class="progress-bar-background">
    <div class="progress-indicator"></div>
  </div>
</div>
`;
